utils/auto_update.py

```
@@ -365,102 +332,103 @@
-                    print("\nInvalid input. Please press Y or N.", end="", flush=True)
-                    start_time = time.time() # Reset timer on invalid input
-        
-        print() # Newline after input or timeout
+            try:
+                if msvcrt.kbhit():
+                    user_input = msvcrt.getch().decode('utf-8').lower()
+                    if user_input in ['y', 'n']:
+                        break
+                    else:
+                        print("\nInvalid input. Please press Y or N.", end="", flush=True)
+            except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)